### PR TITLE
Forgotten password

### PR DESCRIPTION
**Pull request by jans510**

### CODE REVIEW

**Comment by halbekanne** [comment]
The tests don't pass. You can fix them but you can also delete the test files if there aren't any useful ones in them (some are auto-generated etc).

File: src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+<p>
+  Du erhälst eine E-Mail mit einem Link, über den du dein Passowrt zurücksetzen
+  kannst.
+</p>
+<p>
+  Bitte prüfe auch den Spam-Ordner auf neue E-Mails. Solltest du keine E-Mail
+  bekommen, kannst du dir die E-Mail erneut zusenden lassen.
+</p>
+
+E-Mail erneu senden
```
Comment (halbekanne) [inline src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html:12]: Misspelling but also shouldn't there be a link with "resend email", I think you're adding this later right?

File: src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<h2>Passwort zurücksetzen</h2>
+
+<p>
+  Du erhälst eine E-Mail mit einem Link, über den du dein Passowrt zurücksetzen
```
Comment (halbekanne) [inline src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html:4]:
```suggestion
  Du erhälst eine E-Mail mit einem Link, über den du dein Passwort zurücksetzen
```

File: src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html (hunk 1 above)
Comment (halbekanne) [inline src/app/features/registration/components/reset-password-email-verification/reset-password-email-verification.component.html:12]:
```suggestion
E-Mail erneut senden
```

File: src/app/features/registration/components/reset-password/reset-password.component.html [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="content-page-outer-container">
+  <div class="content-page-inner-container content-page-inner-container-narrow">
+    <h2>Passwort zurücksetzen</h2>
+    <p>
+      Bitte gib die email Adresse an, mit der du dich bei uns registriert hast.
```
Comment (halbekanne) [inline src/app/features/registration/components/reset-password/reset-password.component.html:5]:
```suggestion
      Bitte gib die E-Mail Adresse an, mit der du dich bei uns registriert hast.
```

File: src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import { APIError } from '@shared/http/response.types';
+
+@Component({
+  selector: 'app-send-verification-link-form',
+  templateUrl: './send-verification-link-form.component.html',
+  styleUrls: ['./send-verification-link-form.component.scss'],
+})
+export class SendVerificationLinkFormComponent implements OnInit {
+  @Input() submitText?: string;
+  submitButtonText?: string;
```
Comment (halbekanne) [inline src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts:18]: submitButtonText isn't used in the component.

File: src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+  verificationForm!: UntypedFormGroup;
```
Comment (halbekanne) [inline src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts:19]: Why untyped? Is this only for the first draft and you make it a normal form group later? Typed forms are the new default with newer versions of Angular: https://angular.io/guide/typed-forms

File: src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+  error?: APIError;
```
Comment (halbekanne) [inline src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts:20]: remember to use this in the template later to show an error message

File: src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts (hunk 4 above)
Comment (jans510) [reply src/app/features/registration/components/send-verification-link-form/send-verification-link-form.component.ts:18]: I exchanged it by submitText. I'll remove it

File: src/app/shared/http/error-messages.ts [hunk 7]
```diff
@@ -9,7 +9,9 @@ export const errorMessages: Record<APIError, string> = {
   EMAIL_NOT_VALIDATED:
     'Die E-Mail Adresse wurde noch nicht verifiziert.\nBitte schaue in deinem E-Mail Postfach nach.',
   NOT_AVAILABLE: 'Der Benutzername ist leider schon vergeben.',
-  VALIDATION_ERROR: '',
+  VALIDATION_ERROR: 'Bei der Validierung der Daten ist ein fehler aufgetreten',
```
Comment (halbekanne) [inline src/app/shared/http/error-messages.ts:12]:
```suggestion
  VALIDATION_ERROR: 'Bei der Validierung der Daten ist ein Fehler aufgetreten',
```

File: src/app/shared/http/error-messages.ts [hunk 8]
```diff
@@ -9,7 +9,9 @@ export const errorMessages: Record<APIError, string> = {
...
+  CODE_VALIDATION_ERROR:
+    'Der von dir benutze Link ist nicht gültig, bitte fordere einen Neuen an.',
```
Comment (halbekanne) [inline src/app/shared/http/error-messages.ts:14]:
```suggestion
    'Der von dir benutzte Link ist nicht gültig, bitte fordere einen neuen an.',
```

**Review by halbekanne — COMMENTED** [review]
Bis auf die Tippfehler alles ok soweit. Layout werde ich dann noch ein wenig schauen wie besprochen.

**Review by halbekanne — APPROVED** [review]

**Pull request merged**